cnf-tests/testsuites/e2esuite/knmstate/knmstate_sriov.go

```diff
@@ -0,0 +1,275 @@
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring an IPv4 address on Physical Function via NMState")
+			nodeNetworkConfigPolicy := nmstatev1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-sriov-device-" + testDevice.Name},
+				Spec: nmstateshared.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": node,
+					},
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(nodeNetworkConfigPolicy.Name)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			By("Verify Virtual Functions have been correctly configured")
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}).
+				WithPolling(5*time.Second).
+				WithTimeout(10*time.Minute).
+				Should(Equal(int64(10)), "the number of the observed VFs is not that same as the one sriov created")
+
+			By("Verify Physical Function IP address has been correctly configured")
+			Eventually(func(g Gomega) {
+				out, err := ipAddrShow(node, testDevice.Name)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(out).To(ContainSubstring("192.0.2.2"))
+				g.Expect(out).To(ContainSubstring("UP"))
+			}).
+				WithPolling(5 * time.Second).
+				WithTimeout(1 * time.Minute).
+				Should(Succeed())
+		})
+	})
+})
+
+func isKnmstateOperatorInstalled() bool {
+	list := nmstatev1.NMStateList{}
+	err := client.Client.List(context.Background(), &list)
+	if k8serrors.IsNotFound(err) {
+		return false
+	}
+
+	Expect(err).ToNot(HaveOccurred())
+	return true
+}
+
+func ipAddrShow(node, intf string) (string, error) {
+	nodeObj, err := client.Client.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("can't get node[%s]: %w", node, err)
+	}
+
+	cmd := []string{"chroot", "/rootfs", "ip", "-brief", "address", "show", "dev", intf}
```

Comment:
The chroot is not needed I think, the container lands in the host networking so ip command should work.
Also just fyi in case you did no know, the ip command has `json` param `ip --json ` in case helps (but here brief and ContainerString is just fine)